dashboard/app/models/user_level.rb

```diff
@@ -138,15 +130,29 @@ def before_unsubmit
     end
   end
 
-  def has_autolocked?(stage)
-    return false unless stage.lockable?
-    unlocked_at && unlocked_at < AUTOLOCK_PERIOD.ago
+  # `locked` is a virtual attribute because it relies on `unlocked_at` to
+  # automatically return `true` after `AUTOLOCK_PERIOD`
+  def locked
+    unlocked_at.nil? || unlocked_at < AUTOLOCK_PERIOD.ago
   end
 
-  def locked?(stage)
+  def locked=(val)
```

Comment:
What is `val`? Can we use a more descriptive name here?

Reply:
This is just a setter for a virtual attribute (boolean), so will just be true or false. Would switching it to "bool" be clearer?

Reply:
Ah I see - maybe `isLocked` or `isUnlocked`?

Reply:
We wanted it to match the formatting of the rest of the table, so that it can be interacted with in the same way (aka submitted, readonly_answers, etc). Also, the getter and setter have to have the same name and this formatting (setter is [name]=(val), getter is just [name]) for it to take on the *special properties* of a virtual attribute.

Reply:
Ah got it, I didn't realize how virtual attributes worked. I still think it would be helpful to rename `val`

Reply:
I totally misunderstood your comment earlier^, sorry! Yes- will rename val to isLocked. Thanks!